sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1119,11 +1114,18 @@ class GroupedAggregations extends Arm {
       boundOrderSpec: Seq[SortOrder],
       orderByPositions: Array[Int],
       partByPositions: Array[Int],
-      inputCb: ColumnarBatch,
+      inputSpillable: SpillableColumnarBatch,
       outputColumns: Array[cudf.ColumnVector]): Unit = {
-    doRunningWindowOptimizedAggs(isRunningBatched, partByPositions, inputCb, outputColumns)
-    doRowAggs(boundOrderSpec, orderByPositions, partByPositions, inputCb, outputColumns)
-    doRangeAggs(boundOrderSpec, orderByPositions, partByPositions, inputCb, outputColumns)
+    withRetryNoSplit(inputSpillable) { attempt =>
```

Comment:
How are `outputColumns` cleaned up on retry?

Reply:
Ah ok, so you can have multiple window definitions, and any of them could fail, leaking some output columns in the process.

I will need to split things up a bit to handle this correctly.

Reply:
Addressed here: https://github.com/NVIDIA/spark-rapids/pull/7824/commits/4091a40c20d5e2cc6dd9ed20725eae754e1d1159. Glad you found this